Project-Aurora/Project-Aurora/Settings/VariableRegistry.cs

```diff
@@ -97,116 +94,36 @@ public void SetVariable(object newvalue)
-                        case BindableNumber<float> b:
-                            Value = b.Value;
-                            b.ValueChanged += _ => ValueChanged();
-                            break;
-                        case BindableNumber<int> b:
-                            Value = b.Value;
-                            b.ValueChanged += _ => ValueChanged();
-                            break;
-                        case BindableNumber<long> b:
-                            Value = b.Value;
-                            b.ValueChanged += _ => ValueChanged();
-                            break;
-                        case BindableColor b:
-                            Value = b.Value;
-                            b.ValueChanged += _ => ValueChanged();
-                            break;
-                        case Bindable<string> b:
-                            Value = b.Value;
-                            b.ValueChanged += _ => ValueChanged();
-                            break;
-                    }
-        }
-
-        internal void ValueChanged()
-        {
-            switch (Bindable)
+            Title = variableRegistryItem.Title;
+            Remark = variableRegistryItem.Remark;
+            Min = variableRegistryItem.Min;
+            Max = variableRegistryItem.Max;
+            Bindable = variableRegistryItem.Bindable;
+            var typ = Value.GetType();
+            var defaultType = variableRegistryItem.Default.GetType();
+            if (!defaultType.Equals(typ) && typ.Equals(typeof(long)) && defaultType.IsEnum)
+                Value = Enum.ToObject(defaultType, Value);
+            else if (!defaultType.Equals(typ) && Value.GetType().Equals(typeof(long)) && TypeUtils.IsNumericType(defaultType))
+                Value = Convert.ChangeType(Value, defaultType);
+            else if (Value == null && !defaultType.Equals(typ))
+                Value = variableRegistryItem.Default;
+            Flags = variableRegistryItem.Flags;
+            if (Bindable != null)
             {
-                case Bindable<bool> b:
-                    Value = b.Value;
-                    break;
-                case BindableNumber<double> b:
-                    Value = b.Value;
-                    break;
-                case BindableNumber<float> b:
-                    Value = b.Value;
-                    break;
-                case BindableNumber<int> b:
-                    Value = b.Value;
-                    break;
-                case BindableNumber<long> b:
-                    Value = b.Value;
-                    break;
-                case BindableColor b:
-                    Value = b.Value;
-                    break;
-                case Bindable<string> b:
-                    Value = b.Value;
-                    break;
+                Value = (Bindable as Bindable<object>).Value;
+                (Bindable as Bindable<object>).ValueChanged += _ => Value = (Bindable as Bindable<object>).Value;
```

Comment:
As before, you cannot cast a generic type to a different generic type even if the type parameter of one extends the type param of the other.

Reply:
https://github.com/antonpup/Aurora/pull/1757#discussion_r336582007